File: archive/largest-perimeter-triangle.js
const largestPerimeter = function (nums) {
  nums = nums.sort((a, b) => b - a)

  for (let i = 0; i < nums.length - 2; i++) {
    if (nums[i] < nums[i + 1] + nums[i + 2]) {
      return nums[i] + nums[i + 1] + nums[i + 2]
    }
  }

  return 0
};

console.log(largestPerimeter([1, 2, 1, 10]))

File: largest-perimeter-triangle.go
func largestPerimeter(nums []int) int {
    sort.SliceStable(nums, func (i int, j int) bool {
        return nums[i] > nums[j]
    })

    for i:=0; i < len(nums) - 2; i++ {
        if (nums[i] < nums[i + 1] + nums[i + 2]) {
            return nums[i] + nums[i + 1] + nums[i + 2]
        }
    }

    return 0
}